### PR TITLE
feat(builder): improve log info of inspectConfig method

### PR DESCRIPTION
**Pull request by chenjiahan**
# PR Details

## Description

- Improve log info of inspectConfig method.
- Added e2e test for inspect config.
- Adjust the orders of resolve.extensions config.

Before: 

截屏2022-10-14 下午3 10 29

After: 

截屏2022-10-14 下午3 14 37

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] I have updated changeset
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/e2e/webpack-builder/inspect-config/inspectConfig.test.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    writeToDisk: true,
+  });
+
+  expect(fs.existsSync(builderConfig)).toBeTruthy();
+  expect(fs.existsSync(webpackConfig)).toBeTruthy();
+  expect(fs.existsSync(webpackNodeConfig)).toBeTruthy();
+
+  fs.removeSync(builderConfig);
+  fs.removeSync(webpackConfig);
+  fs.removeSync(webpackNodeConfig);
```
Comment (Asuka109) [inline tests/e2e/webpack-builder/inspect-config/inspectConfig.test.ts:49]: stub builder 每次 distPath 是在不同的临时目录，应该不需要手动删除的

File: packages/builder/builder-webpack-provider/src/stub/builder.ts [hunk 2]
```diff
@@ -138,6 +139,18 @@ export async function createStubBuilder(options?: StubBuilderOptions) {
...
+    const { inspectConfig: inspectConfigImpl } = await import(
+      '../core/inspectConfig'
+    );
+    return inspectConfigImpl({
+      context,
+      pluginStore,
+      builderOptions,
+      inspectOptions,
+    });
+  };
```
Comment (Asuka109) [inline packages/builder/builder-webpack-provider/src/stub/builder.ts:152]: inspectConfig 为啥要放进 stub builder 嘞

File: tests/e2e/webpack-builder/inspect-config/inspectConfig.test.ts (hunk 1 above)
Comment (chenjiahan) [reply tests/e2e/webpack-builder/inspect-config/inspectConfig.test.ts:58]: 我跑的时候一个 case 影响了另一个 case，所以手动跑了删除。

现在把 inspectConfig 从 stubBuilder 里移除了，换成了标准的 builder API，removeSync 应该还要留着

File: tests/e2e/webpack-builder/inspect-config/inspectConfig.test.ts [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  const builder = await createBuilder(
+    builderWebpackProvider({
+      builderConfig: {},
+    }),
+    {
+      entry: {
+        index: path.resolve('./src/index.js'),
+      },
+    },
+  );
```
Comment (Asuka109) [inline tests/e2e/webpack-builder/inspect-config/inspectConfig.test.ts:24]: 其实我是想着这个能力可能可以放到 primary builder 里，但是现在取消这个部分了我想想有啥别的办法复用代码吧

**Review by Asuka109 — APPROVED** [review]

File: tests/e2e/webpack-builder/inspect-config/inspectConfig.test.ts (hunk 3 above)
Comment (chenjiahan) [reply tests/e2e/webpack-builder/inspect-config/inspectConfig.test.ts:24]: 可以滴

**Pull request merged**